_inc/lib/admin-pages/class.jetpack-react-page.php

```diff
@@ -136,10 +136,16 @@ function page_admin_scripts() {
 			return; // No need for scripts on a fallback page
 		}
 
+		$script_deps_path    = JETPACK__PLUGIN_DIR . '_inc/build/admin.deps.json';
+		$script_dependencies = file_exists( $script_deps_path )
+			? json_decode( file_get_contents( $script_deps_path ) )
+			: array();
+		$script_dependencies[] = 'wp-polyfill';
+
 		wp_enqueue_script(
 			'react-plugin',
 			plugins_url( '_inc/build/admin.js', JETPACK__PLUGIN_FILE ),
-			array( 'wp-i18n', 'wp-polyfill' ),
+			$script_dependencies,
```

Comment:
Do you know  why `wp-i18n` was a dependency here?

Looks  like admin panel  is using `import { translate as __ } from 'i18n-calypso';`  instead. 🤔

Reply:
Looks like it came from https://github.com/Automattic/jetpack/pull/11225

The new `admin.deps.json` doesn't include it so seems like it wasn't needed:
```json
["lodash","moment","react","react-dom"]
```

Reply:
> Do you know why `wp-i18n` was a dependency here?
> 
> Looks like admin panel is using `import { translate as __ } from 'i18n-calypso';` instead. 🤔

TL;DR - pretty sure it's safe to remove that dependency.

---

I suspect it's related to this:

https://github.com/Automattic/jetpack/blob/17221d7fbd37799508f5b1f43c41e46b7dea2a0b/_inc/lib/admin-pages/class.jetpack-react-page.php#L147

That ends up hitting [this](https://github.com/WordPress/WordPress/blob/49351e0578797a70c592802674ff2b120c367f21/wp-includes/class.wp-scripts.php#L498-L522) which adds the dependency to the script:

```php
	public function set_translations( $handle, $domain = 'default', $path = null ) {
		if ( ! isset( $this->registered[ $handle ] ) ) {
			return false;
		}

		/** @var \_WP_Dependency $obj */
		$obj = $this->registered[ $handle ];

		if ( ! in_array( 'wp-i18n', $obj->deps, true ) ) {
			$obj->deps[] = 'wp-i18n';
		}

		return $obj->set_translations( $domain, $path );
	}
```

which is necessary, because [printing the locales depends](https://github.com/WordPress/WordPress/blob/60670888486617c6eeba7f6d038dd821941bbda3/wp-includes/class.wp-scripts.php#L549-L555) on `wp.i18n`:

```
		$output = <<<JS
( function( domain, translations ) {
	var localeData = translations.locale_data[ domain ] || translations.locale_data.messages;
	localeData[""].domain = domain;
	wp.i18n.setLocaleData( localeData, domain );
} )( "{$domain}", {$json_translations} );
JS;
```

Reply:
You both were right about this one - this was intended to be the Settings area using wp.i18n, but it didn't happen yet. So I have pushed a new commit that would remove `set_script_translations` entirely, so that we wouldn't need `wp-i18n` as a dependency.